pkg/rancher-desktop/preload/extensions.ts

```diff
@@ -5,21 +5,266 @@
 
 import Electron from 'electron';
 
+import type { SpawnOptions } from '@pkg/main/extensions/types';
+import clone from '@pkg/utils/clone';
 import { ipcRenderer } from '@pkg/utils/ipcRenderer';
 
 /* eslint-disable import/namespace -- that rule doesn't work with TypeScript type-only imports. */
 import type { v1 } from '@docker/extension-api-client-types';
 
+// We use a bunch of symbols for names of properties we do not want to reflect
+// over.
+const stream = Symbol('stream');
+const stdout = Symbol('stdout');
+const stderr = Symbol('stderr');
+const id = Symbol('id');
+
+/** execProcess holds the state associated with a v1.ExecProcess. */
+interface execProcess {
+  /** The identifier for this process. */
+  [id]: string;
+  [stdout]: string;
+  [stderr]: string;
+  [stream]: v1.ExecStreamOptions;
+}
+
+/**
+ * The identifier for the extension (the name of the image).
+ */
+const extensionId = decodeURIComponent((location.href.match(/:\/\/([^/]+)/)?.[1] ?? '').replace(/(..)/g, '%$1'));
```

Comment:
I don't understand this. You're essentially taking the domain+port part of `location.href` and inserting a percent character before every two characters -- is this `domain+port` thing  a hex-encoding ?  Of what?

Reply:
Hmm, I only intended to get the domain part (so that's a bug), but yes, the domain is the hex-encoded string of the extension id (i.e. the image name). The hex encoding is describe where we register the protocol handler:
https://github.com/rancher-sandbox/rancher-desktop/blob/4d8f2bf1c5162b9486ebf4c14f085368f4d2be59/pkg/rancher-desktop/utils/protocols.ts#L91-L98